src/main/kotlin/no/nav/familie/ef/sak/beregning/skolepenger/BeregningSkolepengerService.kt

```diff
@@ -1,41 +1,88 @@
 package no.nav.familie.ef.sak.beregning.skolepenger
 
+import no.nav.familie.ef.sak.felles.util.skoleår
 import no.nav.familie.ef.sak.infrastruktur.exception.brukerfeilHvis
-import no.nav.familie.ef.sak.vedtak.dto.InnvilgelseSkolepenger
-import no.nav.familie.ef.sak.vedtak.dto.UtgiftsperiodeSkolepengerDto
+import no.nav.familie.ef.sak.infrastruktur.exception.brukerfeilHvisIkke
+import no.nav.familie.ef.sak.vedtak.dto.SkoleårsperiodeSkolepengerDto
 import org.springframework.stereotype.Service
+import java.time.Year
+import java.util.UUID
+
+private val maksbeløpPerSkoleår = 68_000
 
 @Service
 class BeregningSkolepengerService {
 
-    fun beregnYtelse(innvilgelse: InnvilgelseSkolepenger): List<BeløpsperiodeSkolepengerDto> {
-        return beregnYtelse(innvilgelse.perioder)
+    fun beregnYtelse(
+        utgiftsperioder: List<SkoleårsperiodeSkolepengerDto>,
+        behandlingId: UUID
+    ): BeregningSkolepengerResponse {
+        return beregnYtelse(utgiftsperioder)
     }
 
-    fun beregnYtelse(utgiftsperioder: List<UtgiftsperiodeSkolepengerDto>): List<BeløpsperiodeSkolepengerDto> {
-        validerGyldigePerioder(utgiftsperioder)
-        validerFornuftigeBeløp(utgiftsperioder)
+    fun beregnYtelse(
+        perioder: List<SkoleårsperiodeSkolepengerDto>,
+    ): BeregningSkolepengerResponse {
+        validerGyldigePerioder(perioder)
+        validerFornuftigeBeløp(perioder)
+        validerSkoleår(perioder)
 
-        return utgiftsperioder.map {
-            BeløpsperiodeSkolepengerDto(
-                it.tilPeriode(),
-                beløp = 100, // TODO håndtere senere
-                BeregningsgrunnlagSkolepengerDto(it.studietype, it.utgifter, it.studiebelastning)
-            )
-        }
+        val perioder = beregnSkoleårsperioder(perioder)
+        return BeregningSkolepengerResponse(perioder)
```

Comment:
skulle vi hatt en `validerUnderMaksBeløp` etter at vi har beregnet perioder med stønader?

Reply:
Det gjøres i `validerFornuftigeBeløp` (trakk ut den nå i egen metode) som gjøres per skoleår, og ikke når selve utbetalingen gjøres
Dette for å kunne støtte at man innvilget en delperiode i neste skoleår, men som betales i dette skoleår, dvs at man då har søttte for å innvilge studier for 18 måneder som går over beløpet for inneværende studieår